contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -280,112 +180,107 @@ contract IpRoyaltyVault is IIpRoyaltyVault, ERC20SnapshotUpgradeable, Reentrancy
+            $.claimerInfo[token][from] = ($.poolInfo[token] * (balanceOf(from) - amount)) / totalSupply;
+        }
 
-        // if the ip is tagged, then the unclaimed royalties are unavailable until the dispute is resolved
-        if (DISPUTE_MODULE.isIpTagged($.ipId)) return 0;
+        super._update(from, to, amount);
+    }
 
-        uint256 balance = balanceOfAt(account, snapshotId);
-        uint256 totalSupply = totalSupplyAt(snapshotId);
-        uint256 claimableToken = $.claimableAtSnapshot[snapshotId][token];
-        return $.isClaimedAtSnapshot[snapshotId][account][token] ? 0 : (balance * claimableToken) / totalSupply;
+    function _clearPendingRewards(address claimer, address token) internal returns (uint256 pending) {
+        pending = _claimableRevenue(claimer, token);
+        if (pending > 0) {
+            emit RevenueTokenClaimed(claimer, token, pending);
+            IERC20(token).safeTransfer(claimer, pending);
+        }
     }
 
-    /// @notice Adds a new revenue token to the vault
-    /// @param token The address of the revenue token
-    function _updateVaultBalance(address token, uint256 amount) internal {
+    function _claimRevenueOnBehalf(address claimer, address[] memory tokenList) internal returns (uint256[] memory) {
         IpRoyaltyVaultStorage storage $ = _getIpRoyaltyVaultStorage();
 
-        if (!ROYALTY_MODULE.isWhitelistedRoyaltyToken(token))
-            revert Errors.IpRoyaltyVault__NotWhitelistedRoyaltyToken();
-        if (amount == 0) revert Errors.IpRoyaltyVault__ZeroAmount();
+        if (ROYALTY_MODULE.isIpRoyaltyVault(claimer) && msg.sender != claimer)
+            revert Errors.IpRoyaltyVault__VaultsMustClaimAsSelf();
 
-        $.tokens.add(token);
-        $.pendingVaultAmount[token] += amount;
+        if (IP_ASSET_REGISTRY.isWhitelistedGroupRewardPool(claimer) && msg.sender != GROUPING_MODULE)
+            revert Errors.IpRoyaltyVault__GroupPoolMustClaimViaGroupingModule();
 
-        emit RevenueTokenAddedToVault(token, amount);
+        uint256[] memory claimedAmounts = new uint256[](tokenList.length);
+        for (uint256 i = 0; i < tokenList.length; i++) {
+            claimedAmounts[i] = _clearPendingRewards(claimer, tokenList[i]);
+            if (claimedAmounts[i] == 0) revert Errors.IpRoyaltyVault__NoClaimableTokens();
+            $.claimerInfo[tokenList[i]][claimer] += claimedAmounts[i];
+        }
+
+        return claimedAmounts;
+    }
+
+    function _claimableRevenue(address claimer, address token) internal view returns (uint256) {
+        // accBalance // accumulate revenue tokens in the vault
+        // totalSupply = totalSupply() // totalSupply of RoyaltyTokens of the vault (IpRoyaltyVault)
+        // accBalancePerShare = accBalance / totalSupply
+        // userAmount = balanceOf(user) // user amount of RoyaltyTokens (IpRoyaltyVault),
+        // means how many share the user has
```

Comment:
means how many share the user has

could be

means how much share the user has

Reply:
updated.